addons/binding/org.openhab.binding.tivo/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,152 @@
+			</parameter>
+			<parameter name="pollForChanges" type="boolean">
+				<default>true</default>
+				<required>true</required>
+				<label>Poll for Channel Changes</label>
+				<description>Check TiVo for channel changes.  Enable if openHAB and a physical remote control (or other services use the Remote Control Protocol) will be used. <![CDATA[ <b> ]]>Default: True (Enabled)<![CDATA[ </b> ]]></description>
+			</parameter>
+			<parameter name="pollInterval" type="integer" max="65535" min="3">
+				<default>10</default>
+				<required>true</required>
+				<label>Polling Interval (Seconds)</label>
+				<description>Number of seconds between polling jobs to update status information from the TiVo. <![CDATA[ <b> ]]>Default: 10<![CDATA[ </b> ]]></description>
+			</parameter>
+			<parameter name="cmdWaitInterval" type="integer" max="65535" min="0">
+                <default>200</default>
+                <required>true</required>
+                <label>Command Wait Interval (Milliseconds)</label>
+                <description>Period to wait AFTER a command is sent to the TiVo in milliseconds, before checking that the command has completed. <![CDATA[ <b> ]]>Default: 200<![CDATA[ </b> ]]></description>
+            </parameter>
+            <parameter name="ignoreChannels" type="text">
+                <default></default>
+                <required>false</required>
+                <label>Channels to Ignore</label>
+                <description>Used in channel UP/DOWN operations to avoid invalid channel numbers that are not part of your subscription (these impact the speed of changing channels).  Channels you list in a comma separated list e.g. 109, 111, 999 are skipped/ignored when changing the channel UP or DOWN.<![CDATA[ <br /><br /> ]]>You can also exclude a range of channel numbers by using a hyphen between the lower and upper numbers e.g. 109, 101, 800-850, 999.  New entries are sorted into numerical order when saved.<![CDATA[ <br /><br /> ]]>During normal channel changing operations any invalid channels detected are automatically learnt and added to this list, however the maximum gap for 'auto learning' between valid channels is 10.  Any gap larger than this will not be automatically learnt and the Channel UP/DOWN operation will fail.  If your service has gap(s) larger than 10 channels, you should exclude these manually or use the<![CDATA[ <b> ]]> Perform Channel Scan<![CDATA[ </b> ]]> function to populate this list. <![CDATA[ <br /><br /><b> ]]>Default: NULL (blank)<![CDATA[ </b> ]]></description>
+            </parameter>
+            <parameter name="minChannel" type="integer" max="9999" min="1">
+                <default>100</default>
```

Comment:
It is for Virgin Media (my service provider)!  I guess this will vary from provider to provider but I don't actually know any providers in their main market USA to check.  So I went with what I know.  The min / max are only used in the channel scan and wrap around when doing channel up/down operations.